client/mainChainElasticClient.go

```diff
@@ -1,21 +1,16 @@
 package client
 
 import (
-	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc"
 )
 
 type mainChainElasticClient struct {
-	*elasticClient
+	elasticproc.DatabaseClientHandler
 	indexingEnabled bool
 }
 
 // NewMainChainElasticClient creates a new sovereign elastic client
-func NewMainChainElasticClient(cfg elasticsearch.Config, indexingEnabled bool) (*mainChainElasticClient, error) {
-	esClient, err := NewElasticClient(cfg)
-	if err != nil {
-		return nil, err
-	}
-
+func NewMainChainElasticClient(esClient elasticproc.DatabaseClientHandler, indexingEnabled bool) (*mainChainElasticClient, error) {
```

Comment:
Add check.ifNil for `esClient` and return error

Reply:
added